nautobot/extras/datasources/git.py

```diff
@@ -138,6 +141,14 @@ def ensure_git_repository(repository_record, logger=None, head=None):  # pylint:
     """
 
     from_url, to_path, from_branch = get_repo_from_url_to_path_and_from_branch(repository_record)
+    # We want to check if the repo is already checked out at head. We also want to avoid calling
+    # nautobot.utilities.git.GitRepo.checkout, because it will cause the URL to be rebuilt causing calls to a secrets
+    # backend. As such, if head is None, we can't perform these checks.
+    if head is not None:
```

Comment:
This needs to be reworked per https://github.com/nautobot/nautobot/pull/4844#discussion_r1417672695 doesn't it?